components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -713,7 +713,8 @@ protected void doAuthentication(HttpServletRequest request, HttpServletResponse
         }
 
         String idpName = FrameworkConstants.LOCAL_IDP_NAME;
-        if (context.getExternalIdP() != null && authenticator instanceof FederatedApplicationAuthenticator) {
+        if (context.getExternalIdP() != null && (AuthenticatorType.FEDERATED.equals(authenticator
+                .getAuthenticatorType()) || AuthenticatorType.CUSTOM.equals(authenticator.getAuthenticatorType()))) {
```

Comment:
Don't we need to check for the user here ?

Reply:
At this point, the authentication have not happened yet, therefore, cannot check user type. Also this check is just  to decide, what should be goes to diagnostic log; whether LOCAL and specific idp name based on the authenticator Type (as local authenticators does not have associated IDP).